pkg/cosign/kubernetes/webhook/validator.go

```diff
@@ -241,39 +245,56 @@ func validatePolicies(ctx context.Context, ref name.Reference, defaultKC authn.K
 	authorityErrors := []error{}
 	for p, authorities := range policies {
 		logging.FromContext(ctx).Debugf("Checking Policy: %s", p)
-		// Now the part about having multiple Authority sections within a
-		// policy, any single match will do:
-		// "When multiple authorities are specified, any of them may be used
-		// to source the valid signature we are looking for to admit an image.""
-		authoritiesValidated := 0
-		for _, authority := range authorities {
-			logging.FromContext(ctx).Debugf("Checking Authority: %+v", authority)
-			// TODO(vaikas): We currently only use the defaultKC, we have to look
-			// at authority.Sources to determine additional information for the
-			// WithRemoteOptions below, at least the 'TargetRepository'
-			// https://github.com/sigstore/cosign/issues/1651
-			opts := ociremote.WithRemoteOptions(remote.WithAuthFromKeychain(defaultKC))
-
-			if authority.Key != nil {
-				// Get the key from authority data
-				if authorityKeys, fieldErr := parseAuthorityKeys(ctx, authority.Key.Data); fieldErr != nil {
-					authorityErrors = append(authorityErrors, errors.Wrap(fieldErr, "failed to parse Key values"))
+		sigs, errs := ValidatePolicy(ctx, ref, kc, authorities, remoteOpts...)
+		if len(errs) > 0 {
+			ret[p] = append(ret[p], authorityErrors...)
+		} else {
+			policiesValidated++
+			signatures[p] = append(signatures[p], sigs...)
+		}
+	}
+	return policiesValidated, signatures, ret
+}
+
+// ValidatePolicy will go through all the Authorities for a given image and
+// return a success if at least one of the Authorities validated the signatures.
+// Returns the validated signatures, or the errors encountered.
+func ValidatePolicy(ctx context.Context, ref name.Reference, kc authn.Keychain, authorities []v1alpha1.Authority, _ ...ociremote.Option) ([]oci.Signature, []error) {
+	// If none of the Authorities for a given policy pass the checks, gather
+	// the errors here. If one passes, do not return the errors.
+	authorityErrors := []error{}
+	for _, authority := range authorities {
+		logging.FromContext(ctx).Debugf("Checking Authority: %+v", authority)
+		// TODO(vaikas): We currently only use the defaultKC, we have to look
+		// at authority.Sources to determine additional information for the
+		// WithRemoteOptions below, at least the 'TargetRepository'
+		// https://github.com/sigstore/cosign/issues/1651
+		opts := ociremote.WithRemoteOptions(remote.WithAuthFromKeychain(kc))
```

Comment:
Should we also include the `ociremote.Option` argument from above?

Reply:
Sure, it's never currently used, so just left it in the TODO, but done :)